src/ast/semantic_analyser.cpp

```diff
@@ -254,6 +254,14 @@ void SemanticAnalyser::visit(Call &call)
     }
     call.type = SizedType(Type::integer, 8);
   }
+  else if (call.func == "cgroupid") {
+    if (check_nargs(call, 1)) {
+      if (check_arg(call, Type::string, 0, true)) {
```

Comment:
This `if` is pointless, no? So just having `check_arg(call, …);` should be enough.

Reply:
I was reproducing the coding style from above

Reply:
Appreciate you sticking to the file's coding style, but I think in this case that code is just wrong